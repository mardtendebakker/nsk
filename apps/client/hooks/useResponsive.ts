import { Breakpoint, useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';

export default function useResponsive(
  query: 'up' | 'down' | 'between',
  start?: Breakpoint,
  end?: Breakpoint,
) : boolean {
  const theme = useTheme();

  const mediaUp = useMediaQuery(theme.breakpoints.up(start));

  const mediaDown = useMediaQuery(theme.breakpoints.down(start));

  const mediaBetween = useMediaQuery(theme.breakpoints.between(start, end));

  if (query === 'up') {
    return mediaUp;
  }

  if (query === 'down') {
    return mediaDown;
  }

  if (query === 'between') {
    return mediaBetween;
  }

  return false;
}
