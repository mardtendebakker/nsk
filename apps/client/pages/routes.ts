export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
