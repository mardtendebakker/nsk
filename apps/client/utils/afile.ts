import { AFile } from './axios/models/aFile';

export function buildAFileLink(afile: AFile): string {
  return `https://${afile.unique_server_filename}.s3.amazonaws.com/${afile.discr}/${afile.original_client_filename}`;
}
