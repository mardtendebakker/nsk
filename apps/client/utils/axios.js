import axios from 'axios';

export { AxiosPromise, AxiosResponse, AxiosError } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333' : 'production_url';

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const SUPLIERS_PATH = '';

export default client;
