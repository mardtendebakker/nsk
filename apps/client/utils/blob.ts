export function openBlob(blob: Blob): void {
  const a = document.createElement('a');
  document.body.appendChild(a);
  a.style.cssText = 'display: none';

  const url = window.URL.createObjectURL(blob);

  a.href = url;
  a.target = '_blank';
  a.click();
  window.URL.revokeObjectURL(url);
}
