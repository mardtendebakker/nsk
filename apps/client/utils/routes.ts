export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
export const CUSTOMERS_CONTACTS = '/customers/contacts';
export const CUSTOMERS_CONTACTS_NEW = '/customers/contacts/new';
export const CUSTOMERS_EMAILS_NEW = '/customers/emails/new';
export const CUSTOMERS_EMAILS = '/customers/emails';
export const SUPPLIERS = '/suppliers/:id';
export const ACCOUNT_VERIFICATION = '/account-verification';
export const SETTINGS = '/settings';
export const ORDERS_PURCHASES = '/orders/purchases';
export const ORDERS_SALES = '/orders/sales';
export const STOCKS_PRODUCTS = '/stock/products';
export const STOCKS_REPAIR_SERVICES = '/stock/repair-services';
