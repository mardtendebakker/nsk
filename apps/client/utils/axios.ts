import axios from 'axios';

export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333/api' : process.env.NX_AXIOS_BASE_URL;

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const CUSTOMERS_PATH = '/customers/:id';
export const SUPPLIERS_PATH = '/suppliers/:id';

export default client;

export interface Customer {
  id: number,
  name?: string,
  representative?: string,
  kvk_nr?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  is_partner?: number,
}

export interface Supplier {
  id: number,
  name?: string,
  representative?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  partner?: string,
}
