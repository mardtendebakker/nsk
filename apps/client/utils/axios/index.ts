import axios from 'axios';
import axiosClient from './client';
import refreshToken from './refreshTokenInterceptor';

export {
  CUSTOMERS_PATH,
  SUPPLIERS_PATH,
  SIGN_IN_PATH,
  REFRESH_TOKEN_PATH,
  USER_INFO_PATH,
  SIGN_UP_PATH,
  CONFIRM_ACCOUNT_PATH,
  CHANGE_PASSWORD_PATH,
  SEND_VERIFICATION_CODE_PATH,
  FORGOT_PASSWORD_PATH,
  SALES_ORDERS_PATH,
  PURCHASE_ORDERS_PATH,
  STOCK_PRODUCTS_PATH,
  USERS_PATH,
  STOCK_REPAIR_SERVICES_PATH,
  ORDER_STATUSES_PATH,
  COMPANIES_PATH,
  LOCATIONS_PATH,
  BULK_PRINT_SALES_PATH,
  BULK_PRINT_PURCHASES_PATH,
  ORDERS_ANALYTICS_PATH,
} from './paths';
export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

axiosClient.interceptors.response.use((response) => response, refreshToken);

export const { CancelToken } = axios;

export default axiosClient;
