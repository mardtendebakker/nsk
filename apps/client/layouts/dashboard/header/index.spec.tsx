import { fireEvent, render } from '@testing-library/react';
import { BULK_EMAIL, CONTACTS_CUSTOMERS, MY_TASKS } from '../../../utils/routes';
import Header from './index';
import useResponsive from '../../../hooks/useResponsive';

const mockRouter = {
  get pathname() { return CONTACTS_CUSTOMERS; },
};

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

jest.mock('../../../hooks/useResponsive', () => jest.fn(() => true));

describe('Header', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
  describe('Horizontal menu items', () => {
    it('should highlight the active menu item', () => {
      const { getByText } = render(<Header />);

      expect(getByText('contacts')).toHaveStyle({ fontWeight: 700 });

      expect(getByText('dashboard')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('stock')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('orders')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('bulkEmail')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('logistics')).toHaveStyle({ fontWeight: 400 });
    });

    it('should highlight myTasks menu item', () => {
      jest.spyOn(mockRouter, 'pathname', 'get').mockReturnValue(MY_TASKS);
      const { getByText } = render(<Header />);

      expect(getByText('myTasks')).toHaveStyle({ fontWeight: 700 });

      expect(getByText('contacts')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('dashboard')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('stock')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('orders')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('bulkEmail')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('logistics')).toHaveStyle({ fontWeight: 400 });
    });
  });

  describe('Side nav menu', () => {
    it('should open menu', () => {
      useResponsive.mockReturnValue(false);
      const { getByTestId, queryByText } = render(<Header />);

      expect(queryByText('dashboard')).toBeNull();
      fireEvent.click(getByTestId('openMenuButton'));
      expect(queryByText('dashboard')).not.toBeNull();
    });

    it('should highlight the active menu item', () => {
      useResponsive.mockReturnValue(false);
      jest.spyOn(mockRouter, 'pathname', 'get').mockReturnValue(BULK_EMAIL);
      const { getByText, getByTestId } = render(<Header />);
      fireEvent.click(getByTestId('openMenuButton'));

      expect(getByText('bulkEmail')).toHaveStyle({ fontWeight: 700 });

      expect(getByText('dashboard')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('stock')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('orders')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('contacts')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('logistics')).toHaveStyle({ fontWeight: 400 });
    });

    it('should highlight myTasks menu item', () => {
      useResponsive.mockReturnValue(false);
      jest.spyOn(mockRouter, 'pathname', 'get').mockReturnValue(MY_TASKS);
      const { getByText, getByTestId } = render(<Header />);
      fireEvent.click(getByTestId('openMenuButton'));

      expect(getByText('myTasks')).toHaveStyle({ fontWeight: 700 });

      expect(getByText('contacts')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('dashboard')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('stock')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('orders')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('bulkEmail')).toHaveStyle({ fontWeight: 400 });
      expect(getByText('logistics')).toHaveStyle({ fontWeight: 400 });
    });
  });
});
