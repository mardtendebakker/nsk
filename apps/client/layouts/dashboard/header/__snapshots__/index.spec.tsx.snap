// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1feu265-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1dkyowb-MuiToolbar-root"
    >
      <img
        alt="logo"
        data-nimg="1"
        decoding="async"
        height="9"
        loading="lazy"
        src="/_next/image?url=%2Fassets%2Flogo.jpg&w=128&q=75"
        srcset="/_next/image?url=%2Fassets%2Flogo.jpg&w=64&q=75 1x, /_next/image?url=%2Fassets%2Flogo.jpg&w=128&q=75 2x"
        style="color: transparent;"
        width="50"
      />
      <ul
        class="MuiList-root MuiList-padding css-1e3wvh6-MuiList-root"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              dashboard
            </p>
          </div>
        </a>
        <a
          href="/stock/products"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              stock
            </p>
          </div>
        </a>
        <a
          href="/orders/purchases"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              orders
            </p>
          </div>
        </a>
        <a
          href="/contacts/customers"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1we0tu4-MuiTypography-root"
            >
              contacts
            </p>
          </div>
        </a>
        <a
          href="/bulk-email"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              bulkEmail
            </p>
          </div>
        </a>
        <a
          href="/logistics"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              logistics
            </p>
          </div>
        </a>
      </ul>
      <div
        class="MuiBox-root css-i9gxme"
      />
      <div
        class="MuiBox-root css-70qvj9"
      >
        <a
          href="/my-tasks"
          style="text-decoration: none;"
        >
          <div
            class="MuiBox-root css-1fojr6f"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              myTasks
            </p>
          </div>
        </a>
        <div
          class="MuiBox-root css-1dwbrks"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="Netherlands"
            height="20"
            src="/assets/icons/ic_flag_nl.svg"
            width="30"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root css-16cpzd8"
        />
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1dnh70-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-18ezm2a-MuiTypography-root"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uym7v6-MuiSvgIcon-root"
            data-testid="ChevronRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
